pkg/kubelet/pod_workers.go

```diff
@@ -637,8 +652,9 @@ func (p *podWorkers) UpdatePod(options UpdatePodOptions) {
 		}()
 	}
 
-	// dispatch a request to the pod worker if none are running
-	if !status.IsWorking() {
+	// dispatch a request to the pod worker if none are running and the pod is
+	// not reserved
+	if !status.IsWorking() && !status.IsReserved() {
```

Comment:
Does add ` !status.IsReserved()` here properly? Because if a new static pod with same full name comes,  status.reserved becomes true, then the old pod worker couldn't enter this condtion again, it would hinder managePodLoop

Reply:
An old static pod and a new static pod have different uids, and status is retrieved by the uid.
So old pod's `status.IsReserved` would false.

Reply:
Is it possible that while kubelet restart, the reserved becomes creating runtime pod, but the original running pod becomes reserved, but it's containers are still running and managePodLoop stop

Reply:
I've inspected this behavior. To summary, yes it is possible but it won't have managePodLoop stop.

When restarting the kubelet, a new static pod that should be reserved can enter the UpdatePod first. (in most cases)

However, there is no config for the old static pod but there is only an orphaned runtime pod that should terminate.
And, an orphaned runtime pod is considered a non-static pod so isConflictedStaticPod returns false for the old static pod and then the old static pod is being terminated.

When restarting kubelet, it is possible that two static pods with the same name can be running for a little while. (we may tolerate this...)
However, there is no deadlock.